setup.cfg

```diff
@@ -49,7 +49,7 @@ test =
 ; release scripts and process.
 dev_release =
     towncrier >= 17.4.0
-    pydoctor == 20.7.0
+    pydoctor >= 20.12.1; python_version >= "3.6"
```

Comment:
maybe install for 3.7 and up.
The reasoning is that pydoctor is only used by the release manager ... or the automated release process.

We don't really need to support any python version.... and can save some seconds during the automated tests.

Reply:
The extra isn't just for CI, so I think it should reflect the actual dependencies, not our testing policy. I'm in favor of streamlining CI, since having to wait half an hour for test results is not good. But that should be done by not asking certain tests to be run, rather than pretending they cannot be run.

Also, I don't think we'd get significant speedups from just cutting off a few cases here and there. As is true for optimization in general, we need good profiling tools first, so we can see where the bottlenecks are. Are we not running more in parallel because we don't have the capacity or because the definitions don't allow it? How much time do we spend on actual tests and how much on setting up environments?

One fundamental change we could consider is the matrix approach: we currently test with 6 Python versions (3.5 - 3.9 and pypy3) on 3 operating systems on every run. It might be sufficient to pick a subset of combinations that does include every Python version and every OS, but not every possible combination. That could halve the number of runs for PRs. There could be a full matrix test run on `trunk` on a daily basis to catch the problems that only occur in specific combinations, but I expect those to be rare.

Anyway, that is a much bigger discussion that should be had on the mailinglist.